packages/esm-patient-medications-app/src/medications-summary/medications-summary.component.tsx

```diff
@@ -16,36 +16,41 @@ export default function MedicationsSummary({ patientUuid }: MedicationsSummaryPr
   const { t } = useTranslation();
   const config = useConfig() as ConfigObject;
   const { launchOrderBasket } = useLaunchOrderBasket(patientUuid);
+  const currentDate = new Date();
 
   const {
-    data: activeOrders,
-    error: activeOrdersError,
-    isLoading: isLoadingActiveOrders,
-    isValidating: isValidatingActiveOrders,
-  } = usePatientOrders(patientUuid, 'ACTIVE');
-
-  const {
-    data: pastOrders,
-    error: pastOrdersError,
-    isLoading: isLoadingPastOrders,
-    isValidating: isValidatingPastOrders,
+    data: allOrders,
+    error: error,
+    isLoading: isLoading,
+    isValidating: isValidating,
   } = usePatientOrders(patientUuid, 'any');
 
+  const pastOrders = allOrders?.filter((order) => {
+    if (order.dateStopped) {
+      return order;
+    }
+
+    if (order.autoExpireDate && order.autoExpireDate < currentDate) {
+      return order;
+    }
+  });
+  const activeOrders = allOrders?.filter((order) => !pastOrders.includes(order));
```

Comment:
Thanks for this!

I'd wrap all of this in `useMemo`, e.g.,

```ts
const [pastOrders, activeOrders] = useMemo(() => {
  const currentDate = new Date();  // move to local scope
	// logic here
}, [allOrders]);
```

Also, running two filters on the same list plus the `includes` means this algorithm is something like O(n^3). The solution with a simple for loop and two arrays where each item gets pushed to either of the two arrays would be O(n).

Finally, check the logic around `order.dateStopped`. In particular, there's no guarantee that `dateStopped`, if it exists, is always in the past.

Reply:
Concerning the `order.dateStopped`, whenever I do `order.dateStopped && order.dateStopped < currentDate`. 
Past orders show up in the active orders table. 
![Screenshot from 2023-03-21 22-34-43](https://user-images.githubusercontent.com/104269786/226725112-b22ee020-acca-4a10-b13e-8578a336f521.png)

Reply:
Ok @gracepotma, Ian talked about creating tickets for this, maybe I will up from there.